RdlMigration/ConvertRDL.cs

```diff
@@ -21,6 +22,11 @@ namespace RdlMigration
     /// </summary>
     public class ConvertRDL
     {
+        private string rootFolder;
+        private ConcurrentDictionary<string, string> reportNameMap = new ConcurrentDictionary<string, string>();
```

Comment:
please add comments to code :)

using reportNameMap dictionary we do not upload report Foo if Foo is already found in existing path. However, this does not handle the scenario where Foo was already present in the workspace. In all cases, we do not know whether the 3 Foo reports have the same content or are 3 different reports with same name. Not sure if this is the problem we need to solve through the tool or rather just inform the user about the conflicts.

Reply:
comments - will do! 

cases where Foo is already present in the workspace is handled by existing logic. an error message will be printed notifying the user of a naming conflict due to an existing report in powerBI. we are not uploading reports with duplicate names.

since in PowerBI, subreport paths are handled by ignoring directory paths and only looking at the file name (as per PR description). therefore, yes, by not uploading the correct subreport, the parent report will just link to the report with the matching name in the current workspace. we might need to fix this issue by adding duplicate name logic.

Reply:
@rvpatkar any thoughts on this? should we just go ahead with current logic or should i add the duplicate name feature?

Reply:
not adding duplicate name support in this PR, will do so if customers complain in the future.